R/datamapping-boxwhisker.R

```diff
@@ -0,0 +1,67 @@
+#' @title BoxWhiskerDataMapping
+#' @docType class
+#' @description  Class for Box Whisker Mapping
+#' @export
+BoxWhiskerDataMapping <- R6::R6Class(
+  "BoxWhiskerDataMapping",
+  inherit = XYDataMapping,
+  public = list(
+    quantiles = NULL,
+    outliers = NULL,
+    yMin = NULL,
+    lower = NULL,
+    middle = NULL,
+    upper = NULL,
+    yMax = NULL,
+
+    initialize = function(...,
+                              yMin = NULL,
+                              lower = NULL,
+                              middle = NULL,
+                              upper = NULL,
+                              yMax = NULL,
+                              quantiles = c(0.05, 0.25, 0.5, 0.75, 0.95),
+                              outliers = NULL) {
+      super$initialize(...)
+      self$quantiles <- sort(quantiles)
+      self$outliers <- outliers
+
+      self$yMin <- yMin %||% self$quantiles[1]

```

Comment:
Let's create a method that verifies that the array Quantiles has exactly 5 elements
Also what happens if lower and middle is defined but not upper and yMax. Should we enforce either one method or the other?

Reply:
I created a new check/error: isOfLength and validateIsOfLength
I am not sure if we should or not: since we have a default quantiles, the user can directly overwrite a part of the boxplot without having to fill the 5 elements each time (unless the user modify quantiles directly).
Thoughts ?

Reply:
in theory it's ok if the quantiles array is null if all other values are undefined. 
This is a weird use case indeed. I am not sure what is best.